packages/codemods/src/command.ts

```diff
@@ -1,18 +1,41 @@
-export class CommandParserResult {
-  // path: string = ""; TODO this doesn't work with jest; why?
-  constructor(public shouldExit = false, public path = '') {}
+import yarr from 'yargs';
+import { Maybe, Nothing } from './maybe';
+export interface CommandParserResult {
+  shouldExit?: boolean;
+  stringFilter: Maybe<string[]>;
+  regexFilter: Maybe<string[]>;
+}
+export interface Commands {
+  help?: boolean;
+  filter?: string;
 }
 
 export class CommandParser {
-  constructor(private logger = console.log) {}
-
-  public parseArgs(argv: string[]) {
-    if (argv.slice(2).indexOf('--help') >= 0) {
-      this.logger(['Usage:', [argv[0], argv[1], '<path in project>'].join(' ')].join('\n'));
-
-      return new CommandParserResult(true);
+  public parseArgs(passedIn: string[]): CommandParserResult {
+    const parsed = yarr([])
+      .help()
+      .exitProcess(false)
+      .option('sfilter', {
+        alias: 'sf',
+        type: 'string',
+        array: true,
+        description: 'pass a list of strings to filter by mod name. Does substring search',
+      })
+      .option('rfilter', {
+        alias: 'rf',
+        array: true,
+        type: 'string',
+        description: 'pass a list of regex to filter by mod name',
+      })
+      .parse(passedIn);
+    if (parsed.help) {
+      return { shouldExit: true, regexFilter: Nothing(), stringFilter: Nothing() };
     }
 
-    return new CommandParserResult(false, argv[2]);
+    return {
+      shouldExit: false,
+      stringFilter: Maybe<string[]>(parsed.sfilter),
+      regexFilter: Maybe<string[]>(parsed.rfilter),
```

Comment:
If both `sfilter` and `rfilter` are passed I imagine things might be complicated downstream. Maybe have the command parser do the right thing and just pass a `filter` as a result?